### PR TITLE
Karen/unit details housing locator

### LINKED ISSUE #52: Simple Process Tracking: Unit Details - Housing Locator View
# Overview

Implement the Unit Details view that only HLs see 

## Frontend

Implement the HL view ([Figma](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=286%3A16755&mode=dev)):

- Edit button (just the button itself)
- Landlord information

Ignore the:
- "Change Availability" button
- Referrals table
- Change Availability popup
- Edit flow
- Approval flow

## Backend

n/a

## Acceptance Criteria
- [x] HL view implemented
- [x] Fix font 
- [x] Add navbar to the Unit Details page

### PR DESCRIPTION
**Pull request by PoliteUnicorn**
## Tracking Info

Resolves #52

## Changes
I fixed the spacing and font issues in the UnitDetails page, and I added the navigation bar at the top of the unit details page. 
I also created an edit button and added the landlord information. The view depends on whether the user is a housing locator or not. The edit button also sets the isEditing variable but the form view hasn't been implemented yet.

## Testing
I tested this out on the dummy units created in the database, and switched my role as housing locator to true/false to test that the page views did change. I also used some console.log() statements to test that the edit button was changing the variable correctly.

## Confirmation of Change
**Housing Locator view:** 
Screenshot 2024-03-06 at 6 26 38 PM
**Normal View**
Screenshot 2024-03-06 at 6 28 50 PM

Questions I have:
When displaying the phone number of the landlord, how should it be formatted? Should I add parentheses and dashes into the number or will it have them already?

### CODE REVIEW

**Review by petabite — CHANGES_REQUESTED** [review]
Nice work! Just one small thing: Can we remove the bullet points for the phone number / email of the landlord. See [designs](https://www.figma.com/file/06vEDN5gtlHlC5hxNXMb4g/USHS-Designs?node-id=286%3A16755&mode=dev).

**Comment by petabite** [comment]
> When displaying the phone number of the landlord, how should it be formatted? Should I add parentheses and dashes into the number or will it have them already?

Let's just leave it as it is for now. We can change the format in the backend if the client wants to see a formatted phone number.

**Review by PoliteUnicorn — COMMENTED** [review]
I fixed some spacing issues and the landlord information! I think it should be good now but please let me know if anything else needs to be changed!

File: frontend/src/pages/UnitDetails.tsx [hunk 1]
```diff
@@ -187,7 +195,7 @@ export function UnitDetails() {
     <ListText key={rule}>{rule}</ListText>
   ));
 
-  if (dataContext.currentUser?.isHousingLocator) {
+  if (!dataContext.currentUser?.isHousingLocator) {
```
Comment (petabite) [inline frontend/src/pages/UnitDetails.tsx]: Is this an intended change?

**Review by petabite — CHANGES_REQUESTED** [review]
Thanks for fixing the landlord information! I noticed the frontend lint check is failing. Also, I noticed there is a lot of duplicated code. Do you think you can refactor it so that it has all the common elements between the HL and RS view but display just the landlord information for HLs. i.e. instead of having the `if (dataContext.currentUser?.isHousingLocator)`  statement wrap around the entire page, we can just have the statement wrap around just the landlord information (which is the only thing different between the views). We want to eliminate duplicate code as much as possible because if we end up having to change something in one view, we only need to update it in one place instead of two pages, which is good for code maintainability.

**Review by nbpillai — COMMENTED** [review]
the page looks good! but, in general I think it would be cleaner/more consistent to use flex and flex gap to structure everything on the page rather than using margins to add space in some places. (margins can cause problems sometimes when styling other nearby elements)

File: frontend/src/pages/UnitDetails.tsx (hunk 1 above)
Comment (PoliteUnicorn) [reply frontend/src/pages/UnitDetails.tsx]: No I undid it!

**Review by PoliteUnicorn — COMMENTED** [review]
Cleaned up redundant code and used more gap properties instead of margins.

File: backend/src/routes/units.ts [hunk 2]
```diff
@@ -20,4 +20,7 @@ router.get("/", UnitController.getUnitsHandler);
 
 router.delete("/:id", requireHousingLocator, UnitController.deleteUnitsHandler);
 
+router.put("/:id", requireHousingLocator, UnitController.getUnitHandler);
+router.put("/:id/approve", requireHousingLocator, UnitController.getUnitHandler);
+
```
Comment (petabite) [inline backend/src/routes/units.ts]: Oops, this change looks like it's for your new task. Can we revert it and make the changes on a new branch?

**Review by petabite — APPROVED** [review]

**Pull request merged**